src/main/kotlin/cc/hyperium/utils/ResourceManager.kt

```diff
@@ -0,0 +1,9 @@
+package cc.hyperium.utils
+
+import net.minecraft.util.ResourceLocation
+
+class ResourceManager : HashMap<String, ResourceLocation>() {
```

Comment:
Rather than make this extend HashMap, you could make this a service that is backed by a hashmap and has proxy methods to access the backing map.

Reply:
In fact, you could also make it implement just the base Map interface which would make it compatible with some Kotlin extension functions as well as outline what methods you should have.